ci.py

```diff
@@ -68,5 +87,37 @@ def setup_pants_version(test_pants_version: PantsVersion):
     f.writelines(original_pants_ini)
 
 
+@contextmanager
+def setup_python_version(test_python_version: PythonVersion):
```

Comment:
This is a pretty large method... might be good to extract some helpers.

Reply:
Good idea! I refactored it https://github.com/pantsbuild/setup/pull/32/files#diff-358c08c0a1f88070338d41f607d00072R90. Let me know if it could use another refactor.